CMakeLists.txt

```diff
@@ -85,55 +79,21 @@ else()
 	message(STATUS "Localization disabled: Gettext tools (msgfmt) missing")
 endif()
 
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/ced)
-
-if (APPLE)
-	set_property(GLOBAL PROPERTY BLA_VENDOR APPLE)
-	set(ACCELERATE_COMMAND "--enable-accelerate")
-	set(AUBIO_CFLAGS "${CMAKE_C_FLAGS} -iframeworkwithsysroot /System/Library/Frameworks/ -mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET} -isysroot '${CMAKE_OSX_SYSROOT}/'")
-else()
-	set(ACCELERATE_COMMAND "--enable-atlas")
-        set_property(GLOBAL PROPERTY BLA_VENDOR OpenBLAS)
-	set(AUBIO_CFLAGS "${CMAKE_C_FLAGS}")
-endif()
-
-find_package(BLAS REQUIRED MODULE)
-find_package(FFTW3 COMPONENTS single REQUIRED MODULE)
-
-if(NOT FFTW3_FOUND OR NOT BLAS_FOUND)
-	message(ERROR "fftw3 or BLAS required to build aubio, but they cannot be found.")
-endif()
-
-include(ExternalProject)
-ExternalProject_Add(build-aubio-from-sources
-	SOURCE_DIR ${CMAKE_SOURCE_DIR}/3rdparty/aubio/
-        BINARY_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-build/
-        INSTALL_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-install/
-        UPDATE_COMMAND ""
-	TMP_DIR ${CMAKE_BINARY_DIR}/3rdparty/aubio-build/tmp/
-	CONFIGURE_COMMAND ""
-	BUILD_COMMAND ${CMAKE_COMMAND} -E env CFLAGS=${AUBIO_CFLAGS} ${CMAKE_SOURCE_DIR}/3rdparty/aubio/waf
-                  ${ACCELERATE_COMMAND} --disable-tests --notests --verbose
-                  --enable-fftw3f --disable-sndfile --disable-avcodec --disable-double
-                  --disable-samplerate --disable-docs --disable-wavread --disable-wavwrite
-                  --disable-tests --notests --disable-examples --disable-apple-audio
-                  --disable-fat --disable-jack --libdir=lib
-                  --prefix=${CMAKE_BINARY_DIR}/3rdparty/aubio-install/
-                  --top=${CMAKE_SOURCE_DIR}/3rdparty/aubio/
-                  --out=${CMAKE_BINARY_DIR}/3rdparty/aubio-build/ configure build install
-	INSTALL_COMMAND ""
+include(FetchContent)
+FetchContent_Declare(ced-sources
+  GIT_REPOSITORY https://github.com/performous/compact_enc_det.git
+  #https://github.com/google/compact_enc_det.git
+  GIT_TAG        master
+  SOURCE_DIR ced-src
 )
 
-ExternalProject_Get_Property(build-aubio-from-sources INSTALL_DIR)
-set (AUBIO_INSTALL_DIR ${INSTALL_DIR})
-add_library(aubio STATIC IMPORTED)
-set_target_properties(aubio PROPERTIES IMPORTED_LOCATION ${AUBIO_INSTALL_DIR}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}aubio${CMAKE_STATIC_LIBRARY_SUFFIX})
-add_dependencies(aubio build-aubio-from-sources)
+FetchContent_MakeAvailable(ced-sources)
```

Comment:
Instead of listing this here let's add it as a cmake module L82 - L90

Reply:
Yes, this is still really draft, I agree this would be much cleaner (and prevent pollution of common part)